internal/librarian/release_notes.go

```diff
@@ -237,6 +237,46 @@ func findLatestGenerationCommit(repo gitrepo.Repository, state *config.Librarian
 	return res, nil
 }
 
+// groupByIDAndSubject aggregates conventional commits for ones have the same Piper ID and subject in the footer.
+func groupByIDAndSubject(commits []*conventionalcommits.ConventionalCommit) []*conventionalcommits.ConventionalCommit {
+	idToCommits := make(map[string][]*conventionalcommits.ConventionalCommit)
```

Comment:
Can we simplify this to be a `map[string]*conventionalcommits.ConventionalCommit` and just keep appending to the Library-IDs if we have seen this piper-id before?

Reply:
I've tried this approach but it was not as simple as the current one.

You need to check whether the `Footer` map exists and use `fmt.Sprintf` or StringBuilder to append the string.